src/codegen.cpp

```diff
@@ -653,19 +653,30 @@ static AttributeList get_attrs_basic(LLVMContext &C)
                 None);
 }
 
-static AttributeList get_attrs_sext(LLVMContext &C)
+static AttributeList get_attrs_box_float(LLVMContext &C, unsigned nbytes)
 {
     return AttributeList::get(C,
                 AttributeSet(),
-                Attributes(C, {Attribute::NonNull}),
+                AttributeSet(AttributeSet::get(C, {Attribute::getWithDereferenceableBytes(C, nbytes)})
+                .addAttributes(C, Attributes(C, {Attribute::NonNull,}))),
+                None);
+}
+
+static AttributeList get_attrs_box_sext(LLVMContext &C, unsigned nbytes)
+{
+    return AttributeList::get(C,
+                AttributeSet(),
+                AttributeSet(AttributeSet::get(C, {Attribute::getWithDereferenceableBytes(C, nbytes)})
+                .addAttributes(C, Attributes(C, {Attribute::NonNull,}))),
```

Comment:
These probably should be using AttrBuilder. I am also thinking of adding an extra argument to `Attributes`, so that you can use that `initializer_list` constructor version for conciseness, but `AttrBuilder` is basically equivalent.